packages/ariakit-utils/src/__tests__/dom-test.ts

```diff
@@ -0,0 +1,601 @@
+  document.body.innerHTML =
+    '<div id="testNode"><iframe id="testIframe"/></div>';
+  const node = document.getElementById("testNode") as HTMLElement;
+  const iframe = document.getElementById("testIframe") as HTMLIFrameElement;
+
+  expect(isFrame(node)).toEqual(false);
+  expect(isFrame(iframe)).toEqual(true);
+
+  // Reset
+  document.body.innerHTML = initialInnerHTML;
+});
+
+test("isButton", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML = `
+    <div id="testNode">
+      <button id="testButton"/>
+      <input type="button" id="testInputButton"/>
+      <input type="submit" id="testInputSubmit"/>
+      <input type="reset" id="testInputReset"/>
+      <input type="image" id="testInputImage"/>
+      <input type="file" id="testInputFile"/>
+      <input type="color" id="testInputColor"/>
+      <input type="text" id="testInput"/>
+      <div role="button" id="testDivButton"/>
+    </div>
+  `;
+  const node = document.getElementById("testNode") as HTMLElement;
+  const button = document.getElementById("testButton") as HTMLButtonElement;
+  const inputButton = document.getElementById(
+    "testInputButton"
+  ) as HTMLInputElement;
+  const inputSubmit = document.getElementById(
+    "testInputSubmit"
+  ) as HTMLInputElement;
+  const inputReset = document.getElementById(
+    "testInputReset"
+  ) as HTMLInputElement;
+  const inputImage = document.getElementById(
+    "testInputImage"
+  ) as HTMLInputElement;
+  const inputFile = document.getElementById(
+    "testInputFile"
+  ) as HTMLInputElement;
+  const inputColor = document.getElementById(
+    "testInputColor"
+  ) as HTMLInputElement;
+  const input = document.getElementById("testInput") as HTMLInputElement;
+  const divButton = document.getElementById(
+    "testDivButton"
+  ) as HTMLButtonElement;
+
+  expect(isButton(node)).toEqual(false);
+  expect(isButton(button)).toEqual(true);
+  expect(isButton(inputButton)).toEqual(true);
+  expect(isButton(inputSubmit)).toEqual(true);
+  expect(isButton(inputReset)).toEqual(true);
+  expect(isButton(inputImage)).toEqual(true);
+  expect(isButton(inputFile)).toEqual(true);
+  expect(isButton(inputColor)).toEqual(true);
+  expect(isButton(input)).toEqual(false);
+  expect(isButton(divButton)).toEqual(false);
+
+  // Reset
+  document.body.innerHTML = initialInnerHTML;
+});
+
+test("matches", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML = '<div id="testNode" />';
+  const element = document.getElementById("testNode") as HTMLElement;
+
+  expect(matches(element, "div")).toEqual(true);
+
+  // Unsure of a better way to test this
+  // Current method is to create fake element with specific properties matches
+  const matchesFunc = jest.fn(() => true);
+  const fakeElement = { matches: matchesFunc } as any;
+  expect(matches(fakeElement, "div")).toEqual(true);
+  expect(matchesFunc).toHaveBeenCalled();
```

Comment:
Do we need to test `matches` on 207-210? Isn't it being tested already on line 203?

Reply:
I guess you are right. I think I just went and made test explicitly for each of the three `matches` functions. Assuming JSdom continues to have `element.matches` then it is duplicative. I think removing it is probably the right call.